src/Shared/IElementLocation.cs

```diff
@@ -7,15 +7,16 @@
 
 namespace Microsoft.Build.Shared
 {
+    internal interface IElementLocation : IMsBuildElementLocation, ITranslatable { }
+
     /// <summary>
     /// Represents the location information for error reporting purposes.  This is normally used to
     /// associate a run-time error with the original XML.
     /// This is not used for arbitrary errors from tasks, which store location in a BuildXXXXEventArgs.
     /// All implementations should be IMMUTABLE.
-    /// This is not public because the current implementation only provides correct data for unedited projects.
-    /// DO NOT make it public without considering a solution to this problem.
+    /// Any editing of the project XML through the MSBuild API's will invalidate locations in that XML until the XML is reloaded.
     /// </summary>
-    internal interface IElementLocation : ITranslatable
+    public interface IMsBuildElementLocation
```

Comment:
Question: Since this file is shared, can we instead of making this interface public add this file to the project where it is needed?

Reply:
tl;dr;: changed to `internal`

It was meant to be public as it is needed in the property read/write OM. But since that OM is internal for now - let's not expose this one either